dhis-2/dhis-api/src/main/java/org/hisp/dhis/util/SharingUtils.java

```diff
@@ -51,17 +49,64 @@ private SharingUtils()
         throw new UnsupportedOperationException( "utility" );
     }
 
-    public static Set<UserGroupAccess> getDtoUserGroupAccesses( Sharing sharing )
+    public static Set<UserGroupAccess> getDtoUserGroupAccesses( Set<org.hisp.dhis.user.UserGroupAccess> dto,
+        Sharing sharing )
     {
-        return sharing.hasUserGroupAccesses() ? sharing.getUserGroups().values()
-            .stream().map( org.hisp.dhis.user.sharing.UserGroupAccess::toDtoObject ).collect( Collectors.toSet() )
-            : new HashSet<>();
+        if ( !CollectionUtils.isEmpty( dto ) )
+        {
+            return dto;
+        }
+
+        if ( dto == null )
+        {
+            dto = new HashSet<>();
```

Comment:
Couldn't this be `return emptySet()` ? to be more clear of flow and avoid unnecessary creation of `HashSet`?

Reply:
Tried and test failed as below. The reason is that this is called by `BaseIdentifiableObject.getUserAccesses()` which will be called by many services for manipulate those sharing collections.

```Error:  org.hisp.dhis.patch.PatchServiceTest.testEmbeddedObjectCollectionDiff  Time elapsed: 0.303 s  <<< ERROR!
58
java.lang.UnsupportedOperationException
59
	at java.base/java.util.AbstractCollection.add(AbstractCollection.java:267)
60
	at org.hisp.dhis.patch.DefaultPatchService.applyMutation(DefaultPatchService.java:432)
61
	at org.hisp.dhis.patch.DefaultPatchService.applyMutation(DefaultPatchService.java:370)
62
	at org.hisp.dhis.patch.DefaultPatchService.lambda$apply$0(DefaultPatchService.java:116)
63
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
64
	at org.hisp.dhis.patch.DefaultPatchService.apply(DefaultPatchService.java:116)
65
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
66
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
67
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
68
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
69
	at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:344)
70
	at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:198)
71
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163)
72
	at org.springframework.transaction.interceptor.TransactionInterceptor$1.proceedWithInvocation(TransactionInterceptor.java:123)
73
	at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:388)
74
	at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:119)
75
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)
76
	at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:215)
77
	at com.sun.proxy.$Proxy242.apply(Unknown Source)
78
	at org.hisp.dhis.patch.PatchServiceTest.testEmbeddedObjectCollectionDiff(PatchServiceTest.java:354)```